lib/switchtec.c

```diff
@@ -789,40 +783,121 @@ static int read_log_defs(FILE *log_def_file, struct log_defs *defs)
 			continue;
 
-        	num_entries = strtol(tok, NULL, 0);
+		num_entries = strtol(tok, NULL, 0);
 
 		/*
 		 * Skip this module if it has already been done. This can happen
 		 * if the module is duplicated in the log definition file.
 		 */
 		if (mod_defs->mod_name != NULL) {
 			for (i = 0; i < num_entries; i++) {
-				if (!fgets(line, sizeof(line),
-					  log_def_file))
+				if (!fgets(line, sizeof(line), log_def_file))
 					break;
 			}
 			continue;
 		}
 
-        	mod_defs->mod_name = strdup(line);
+		mod_defs->mod_name = strdup(line);
 		mod_defs->num_entries = num_entries;
 		mod_defs->entries = calloc(mod_defs->num_entries,
 					   sizeof(*mod_defs->entries));
 		if (!mod_defs->entries)
-			goto err_free_log_defs;
+			return -1;
 
 		for (i = 0; i < mod_defs->num_entries; i++) {
 			if (fgets(line, sizeof(line), log_def_file) == NULL) {
 				errno = SWITCHTEC_ERR_LOG_DEF_READ_ERROR;
-				goto err_free_log_defs;
+				return -1;
 			}
 
 			mod_defs->entries[i] = strdup(line);
 			if (!mod_defs->entries[i])
-				goto err_free_log_defs;
+				return -1;
 		}
 	}
 
+	return 0;
+}
+
+/**
+ * @brief Read a mailbox log definition file and store the definitions
+ * @param[in] log_def_file - log definition file
+ * @param[out] defs 	   - log definitions
+ * @return 0 on success, negative value on failure
+ */
+static int read_mailbox_log_defs(FILE *log_def_file, struct log_defs *defs)
+{
+	char line[512];
+	struct module_log_defs *mod_defs;
+	int num_entries_alloc;
+
+	/* at least one log definition entry must be allocated */
+	if (defs->num_alloc < 1)
+		return -1;
+
+	/*
+	 * The mailbox log definitions don't keep track of modules. Just store
+	 * everything in the first entry.
+	 */
+	mod_defs = &defs->module_defs[0];
+	mod_defs->num_entries = 0;
+
+	/* allocate some entries */
+	num_entries_alloc = 100;
+	mod_defs->entries = calloc(num_entries_alloc,
+				   sizeof(*mod_defs->entries));
+	if (!mod_defs->entries)
+		return -1;
+
+	while (fgets(line, sizeof(line), log_def_file)) {
+		if (mod_defs->num_entries >= num_entries_alloc) {
+			/* allocate more entries */
+			num_entries_alloc *= 2;
+			mod_defs->entries = realloc(mod_defs->entries,
+						    (num_entries_alloc *
+						     sizeof(*mod_defs->entries)));
+			if (!mod_defs->entries)
+				return -1;			
+		}
+
+		mod_defs->entries[mod_defs->num_entries] = strdup(line);
+		if (!mod_defs->entries[mod_defs->num_entries])
+			return -1;
+
+		mod_defs->num_entries++;
+	}
+
+	return 0;
+}
+
+/**
+ * @brief Read an app log or mailbox log definition file and store the
+ * 	  definitions
+ * @param[in] log_def_file - log definition file
+ * @param[in] log_type     - log type
+ * @param[out] defs 	   - log definitions
+ * @return 0 on success, negative value on failure
+ */
+static int read_log_defs(FILE *log_def_file,
+			 enum switchtec_log_parse_type log_type,
+			 struct log_defs *defs)
+{
+	int ret;
+
+	/* allocate some log definition entries */
+	ret = realloc_log_defs(defs, 200);
```

Comment:
This is confusing me.... read_mailbox_log_defs() never seems to call realloc_log_defs() so presumably it's only actually using one module... But we always allocate 200?

I don't really see any value in this static function, it's probably better to just put the allocation into each of read_app_log_defs() and read_mailbox_log_defs() respectively... they clearly need different things.

Reply:
Sure. I'll remove this function and move the allocation into those ones.